File: lib/UIAlertController+Utils.m
#import "UIAlertController+Utils.h"

#import "NSBundle+Localisation.h"

@implementation UIAlertController (Utils)

- (void)show {
    UIViewController *vc = [[[UIApplication sharedApplication] keyWindow] rootViewController];
    [vc presentViewController:self animated:YES completion:nil];
}

+ (UIAlertController *)likeAlertWithType:(NSString *)type 
                            successBlock:(void (^)(void))successBlock {
    NSString *message = [NSString stringWithFormat:L(@"instap4tch0.alert.double_tap_guard.message"), type];
    UIAlertController* alert = 
    [UIAlertController alertControllerWithTitle:L(@"instap4tch0.generic.warning")
                                        message:message
                                 preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction: [self _yesActionWithSuccessBlock:successBlock]];
    [alert addAction: [self _noAction]];
    return alert;
}

+ (UIAlertAction*)_yesActionWithSuccessBlock:(void (^)(void))successBlock {
    return
    [UIAlertAction actionWithTitle:L(@"instap4tch0.generic.yes")
                             style:UIAlertActionStyleDefault
                            handler:^(UIAlertAction *action){
        successBlock();
    }];
}

+ (UIAlertAction*)_noAction {
    return
    [UIAlertAction actionWithTitle:L(@"instap4tch0.generic.no")
                             style:UIAlertActionStyleDefault
                            handler:nil];
}

@end

File: lib/UIAlertController+Utils.swift
import UIKit

extension UIAlertController {

    @objc
    public func show() {
        let rootVC = UIApplication.shared.keyWindow?.rootViewController
        rootVC?.present(self, animated: true, completion: nil)
    }

    @objc(showOnViewController:)
    public func show(on viewController: UIViewController) {
        viewController.present(self, animated: true, completion: nil)
    }

    @objc(errorAlertForError:)
    public static func errorAlert(for error: Error?) -> UIAlertController {
        let message = error?.localizedDescription ?? L("alert.error.default_message")
        let alert = UIAlertController(title: L("generic.error"),
                                      message: message,
                                      preferredStyle: .alert)
        alert.addAction(UIAlertController.Actions.dismiss())
        return alert
    }

    @objc(localizedAlertWithTitle:message:)
    public static func localizedAlert(withTitle title: String, message: String) -> UIAlertController {
        let alert = UIAlertController(title: L(title),
                                      message: L(message),
                                      preferredStyle: .alert)
        alert.addAction(UIAlertController.Actions.dismiss())
        return alert
    }

    @objc(debugAlertWithItem:)
    public static func debugAlert(with item: AnyObject) -> UIAlertController {
         let alert = UIAlertController(title: "Debug",
                                      message: item.debugDescription,
                                      preferredStyle: .alert)
        alert.addAction(UIAlertController.Actions.dismiss())
        return alert
    }
}